### PR TITLE
Unit tests#138

### PR DESCRIPTION
**Pull request by lvdoorn**
I have added a start to unit testing. So far only modules that do not depend on library functions can only be tested.

### CODE REVIEW

**Comment by borek2** [comment]
Whoooo finally tests! 💃

File: karma.conf.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ module.exports = function (config) {
 
         // list of files to exclude
         exclude: [
+            'src/main.js'
```
Comment (mboom) [inline karma.conf.js]: Why is main excluded?

Comment (lvdoorn) [inline karma.conf.js]: Main launches the standard application, which does not need to happen when only running the tests. Including main will confuse karma.

File: spec/services/OctopeerAPI_test.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+            var parameters = [];
+            expect(opapi.urlBuilder(endpoint, parameters)).toEqual("http://146.185.128.124/api/repositories");
+        });
+
+        it('should build a url with parameters correctly', function () {
+            var opapi = new OctopeerAPI();
+            var endpoint = '/repositories';
+            var parameters = { a: 1, b: 2, c: 3 };
+            expect(opapi.urlBuilder(endpoint, parameters)).toEqual("http://146.185.128.124/api/repositories?a=1&b=2&c=3");
+        });
```
Comment (mboom) [inline spec/services/OctopeerAPI_test.js]: Maybe add a offline / wrong server test?

Comment (borek2) [inline spec/services/OctopeerAPI_test.js]: dont think a build should fail because of a server that is offline

**Pull request merged**